File: Beispiele/Java2Kotlin/src/main/kotlin/javakotlinworkshop/java/LoopDemo.java
package javakotlinworkshop.java;

import java.util.Arrays;

public class LoopDemo {

    public static void main(String[] args) {
        for (int i = 1; i <= 3; i++) {
            System.out.println(i);
        }

        for (int i = 6; i >= 0; i -= 2) {
            System.out.println(i);
        }

        var array = new String[]{"Eins", "Zwei", "Drei"};
        for (var s : array) {
            System.out.println(s);
        }

        var list = Arrays.asList("Eins", "Zwei", "Drei");
        list.forEach(s -> System.out.println(s));

        int i = 0;
        while (i++ < 3) {
            System.out.println(i);
        }
    }
}

File: Beispiele/Java2Kotlin/src/main/kotlin/javakotlinworkshop/kotlin/LoopDemo.kt
package javakotlinworkshop.kotlin

fun main(args: Array<String>) {
    for (i in 1..3) {
        println(i)
    }

    for (i in 6 downTo 0 step 2) {
        println(i)
    }

    val array = arrayOf("Eins", "Zwei", "Drei")
    for (i in array.indices) {
        println(array[i])
    }

    val list = listOf("Eins", "Zwei", "Drei")
    for (i in list) {
        println(i)
    }

    list.forEach { println(it) }

    var i = 0
    while (i++ < 3) {
        println(i)
    }
}